### PR TITLE
feat: display generating mana message properly

### LINKED ISSUE #8391: [Task]: Display `Generating Mana..` message properly
The message says it will refresh in X seconds when it shouldn't, if you've already sent it...

hint: there is a moment in the wallet where the mana and balance are both zero for some reason

Image

### LINKED ISSUE #8417: [Task]: Fix jumping mana generation counting
### Task description

This issue happend when minting
 an NFT, it might happen in other places, so all places should be tested
Also, why dont we display the refresh countdown?

https://github.com/iotaledger/firefly/assets/3624944/13129efb-86d6-42ee-8586-433cb2f82478

### PR DESCRIPTION
**Pull request by cpl121**
Closes #8391 
Closes #8417 

needs to be tested

## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/components/ManaBox.svelte [hunk 1]
```diff
@@ -41,7 +42,7 @@
             requiredTxManaCost = Number(requiredManaForTransaction ?? 0)
 
             const slotsRemaining = Number(splittedError.reverse()[0].replace('`', ''))
-            if (slotsRemaining) {
+            if (slotsRemaining && !secondsRemaining) {
```
Comment (marc2332) [inline packages/desktop/components/ManaBox.svelte:45]: I think this is more readable.
```suggestion
            if (slotsRemaining && secondsRemaining === 0) {
```

**Review by marc2332 — CHANGES_REQUESTED** [review]
Just a small suggestion, it seems to work! ⚡

File: packages/desktop/components/popups/AllotManaPopup.svelte [hunk 2]
```diff
@@ -130,7 +141,9 @@
                     submitHandler={onSubmit}
                     disabled={isBusy}
                 />
-                <ManaBox {transactionInfo} bind:hasEnoughMana />
+                {#if displayManaBox}
+                    <ManaBox {transactionInfo} bind:hasEnoughMana />
+                {/if}
             </div>
             <popup-buttons class="flex flex-row flex-nowrap w-full space-x-4">
                 <Button classes="w-full" outline onClick={onBackClick} disabled={isBusy}
```
Comment (evavirseda) [inline packages/desktop/components/popups/AllotManaPopup.svelte:149]: if there is no amount and no account address, I would disable this button :pray:

**Review by evavirseda — CHANGES_REQUESTED** [review]

**Review by evavirseda — APPROVED** [review]
:rocket:

**Review by begonaalvarezd — APPROVED** [review]

**Pull request merged**